spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubProperties.java

```diff
@@ -129,6 +143,18 @@ public void setExecutorThreads(int executorThreads) {
 		 */
 		private Optional<Integer> parallelPullCount = Optional.empty();
```

Comment:
I'm not sure that this is a good variant for the configuration property.
I mean no `Optional` for them, please.

@snicoll ,
WDYT  here?

Consider to refactor this class properly: https://github.com/spring-projects/spring-framework/wiki/Code-Style#java-source-file-organization

I mean no getters before properties.

Also all the configuration properties must have JavaDocs - for the future metadata for the auto-completion assistant and so on.

Reply:
Are you proposing just leaving things as null instead of Optional.empty to tell that the property wasn't set?

Reply:
That's correct. At least this is what we do in other Spring Boot places for simple types.
And that's why I invited @snicoll to judge us: may be this is really legit to use `Optional` since I see that you have tested it in your changes.

Reply:
Thanks Artem. Using `Optional` on properties [is not really recommended](https://stackoverflow.com/questions/29033518/is-it-a-good-practice-to-use-optional-as-an-attribute-in-a-class) which is the main reason we don't support it.

Reply:
@snicoll Has something changed since this ticket? https://github.com/spring-projects/spring-boot/issues/5110

It looks like you personally added support for Optional there.

Reply:
I guess support of a feature is not necessarily an endorsement of it. :smile: 
I'm with @snicoll regarding not using it in the class attributes though. Can we go back to `@Nullable` or `required=false`?

Reply:
It sounded like you thought it was a good idea enough to confirm to that individual that Spring would be adding support for what he wanted...

Separately, @meltsufin  this isn't related to the Nullable or Autowire(required=false) issue. That was instead covered by the ObjectProvider thing.

Reply:
> It sounded like you thought it was a good idea enough to confirm to that individual that Spring would be adding support for what he wanted...

We can't really enforce our opinions on everything. Using `Optional` on class attributes is a bad idea IMO. Yet, if we support `Optional`, then  we should support that consistently. 

If a property isn't available in the `Environment` it should return `null` and not `Optional.empty()` for consistency with `Environment#getProperty`